apps/finance/contracts/Finance.sol

```diff
@@ -136,6 +136,34 @@ contract Finance is App, Initializable, ERC677Receiver {
         require(_token.transferFrom(msg.sender, address(vault), _amount));
     }
 
+    /**
+     * @dev Deposit for ERC20 tokens using approveAndCall
+     * @param _from Address sending the tokens
+     * @param _amount Amount of tokens sent
+     * @param _token Token being deposited
+     * @param _data Data payload being executed (payment reference)
+     */
+    function receiveApproval(
+        address _from,
+        uint256 _amount,
+        address _token,
+        bytes _data
+    )
+        transitionsPeriod
+        external
+        returns (bool success)
+    {
+        ERC20 token = ERC20(_token);
+        _recordIncomingTransaction(
+            token,
+            _from,
+            _amount,
+            string(_data)
+        );
+        require(token.transferFrom(_from, address(vault), _amount));
+        return true;
```

Comment:
Same as this https://github.com/aragon/aragon-apps/pull/59#discussion_r160099249. Lets make the same in both cases :)